src/components/structures/UserMenu.tsx

```diff
@@ -478,17 +490,7 @@ export default class UserMenu extends React.Component<IProps, IState> {
         >
             <div className="mx_UserMenu_contextMenu_header">
                 {primaryHeader}
-                <AccessibleTooltipButton
-                    className="mx_UserMenu_contextMenu_themeButton"
-                    onClick={this.onSwitchThemeClick}
-                    title={this.state.isDarkTheme ? _t("Switch to light mode") : _t("Switch to dark mode")}
-                >
-                    <img
-                        src={require("../../../res/img/element-icons/roomlist/dark-light-mode.svg")}
-                        alt={_t("Switch theme")}
-                        width={16}
-                    />
-                </AccessibleTooltipButton>
+                {SettingsStore.getValue(UIFeature.ThemeChanging) && themebutton}
```

Comment:
I had brief discussion few days ago. Usually with template definitions , variable definition is used so that it makes code more readable in if statement but We get choice of two: One is to make code readable and make that condition reusable for future but at the same time other decision would be to stick to matrix pattern to avoid merge conflict. I would also incline to your approach Jasen just to avoid Merge Conflict but I had to write everything from scratch which doesnt apply to this case, I would define html as variable and inject when needed. Somewhere around other javascript files I have found matrix using variable definition , so there has not been consistency in their approach.

Reply:
Surendra I agree with the variable approach if we were starting from scratch ... you are always going to get inconsistencies with multiple developers, and more importantly multiple code reviewers